File: 351001/Tereshchenko/RVTaskFull/TweetService/src/main/java/com/tweetservice/configs/restclientconfig/RestClientConfig.java
package com.tweetservice.configs.restclientconfig;

import org.springframework.cloud.client.loadbalancer.LoadBalanced;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;
import org.springframework.web.client.RestClient;

@Configuration
public class RestClientConfig {

    @Bean
    @Primary
    public RestClient.Builder primaryBuilder() {
        return RestClient.builder();
    }

    @Bean
    @LoadBalanced
    public RestClient.Builder balancedBuilder() {
        return RestClient.builder();
    }

    @Bean
    public RestClient restClient(@LoadBalanced RestClient.Builder builder) {
        return builder.build();
    }
}

File: 351003/Soika/entities-app/publisher/src/main/kotlin/com/example/entitiesapp/config/RestClientConfig.kt
package com.example.entitiesapp.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestClient

@Configuration
class RestClientConfig {
    @Bean
    fun discussionClient(): RestClient {
        return RestClient.builder()
            .baseUrl("http://localhost:24130${ApiConfig.API_V1}")
            .build()
    }
}